wsmaster/che-core-api-factory-github/src/main/java/org/eclipse/che/api/factory/server/github/GithubApiClient.java

```diff
@@ -132,6 +137,46 @@ public GithubPullRequest getPullRequest(
         });
   }
 
+  /**
+   * Returns the last commit of the branch.
+   *
+   * <p>GitHub REST API documentation: https://docs.github.com/en/rest/commits/commits
+   *
+   * @param user
+   * @param repository
+   * @param branch
+   * @param authenticationToken
+   * @return the last commit of the branch
+   */
+  public GithubCommit getLastCommit(
+      String user, String repository, String branch, String authenticationToken)
+      throws ScmBadRequestException, ScmItemNotFoundException, ScmCommunicationException,
+          URISyntaxException {
+
+    final URI uri = apiServerUrl.resolve(String.format("./repos/%s/%s/commits", user, repository));
+
+    final URI requestURI =
+        new URI(
+            uri.getScheme(),
+            uri.getAuthority(),
+            uri.getPath(),
+            String.format("sha=%s&page=1&per_page=1", branch),
+            null);
+    HttpRequest request = buildGithubApiRequest(requestURI, authenticationToken);
+    LOG.trace("executeRequest={}", request);
+
+    return executeRequest(
+        httpClient,
+        request,
+        response -> {
+          try {
+            return OBJECT_MAPPER.readValue(response.body(), GithubCommit[].class)[0];
```

Comment:
Are you sure that the firs object from the list represents the latest commit?

Reply:
The array must contain only one object because of
```
"page=1&per_page=1"
```